develop/application-devs/interact/understand-parachain-data.md

```diff
@@ -0,0 +1,388 @@
+
+- Use the [Polkadot Portal](https://polkadot.js.org/apps/#/rpc) to connect to a blockchain or node and select the **state** endpoint and the **getMetadata** method to return the metadata in JSON format.
+- Use the command-line `polkadot-js-api` to call the `state_getMetadata` RPC method to return the metadata as a hex-encoded vector of SCALE-encoded bytes.
+- Use the `subxt metadata` command to download the metadata in JSON, hex, or raw bytes.
+- Use the `sidecar` API and `/runtime/metadata` endpoint to connect to a node and retrieve the metadata in JSON format.
+
+The type information provided by the metadata enables applications to communicate with nodes with different versions of the runtime and across chains that expose different calls, events, types, and storage items.
+The metadata also allows libraries to generate a substantial portion of the code needed to communicate with a given node, enabling libraries like [`subxt`](https://github.com/paritytech/subxt) to generate front-end interfaces that are specific to a target chain.
+
+
+### Using `subxt` for metadata retrieval
+
+One of the quickest ways is to use `subxt`, which you may use to fetch the metadata of any data in a human-readable JSON format: 
+
+```sh
+subxt metadata  --url wss://rpc.polkadot.io --format json > spec.json
+```
+
+Another option is to use the [`subxt` explorer web UI](https://paritytech.github.io/subxt-explorer/#/).
+
+## Client applications and metadata
+
+Client applications use the metadata to interact with the node, parse responses, and format message payloads sent to the node.
+To use the metadata, client applications must use the [SCALE codec library](https://github.com/paritytech/parity-scale-codec?tab=readme-ov-file#parity-scale-codec) to encode and decode RPC payloads.
+Because the metadata exposes how every type is expected to be decoded, applications can send, retrieve, and process application information without manual encoding and decoding.
+
+## Metadata format
+
+Although the SCALE-encoded bytes can be decoded using the `frame-metadata` and [`parity-scale-codec`](https://github.com/paritytech/parity-scale-codec) libraries, there are other tools—such as `subxt` and the Polkadot-JS API—that can convert the raw data to human-readable JSON format.
+
+The types and type definitions included in the metadata returned by the `state_getMetadata` RPC call depend on the metadata version of the runtime.
+In general, the metadata includes the following information:
+
+- A constant that identifies the file as containing metadata.
+- The version of the metadata format used in the runtime.
+- Type definitions for all types used in the runtime and generated by the `scale-info` crate.
+- Pallet information for all of the pallets included in the runtime in the order that they are defined in the `construct_runtime` macro.
+
+The following example illustrates a condensed and annotated section of metadata decoded and converted to JSON:
+
+```json
```

Comment:
For better maintainability, consider using the framework's built-in snippet tool:

1. Store full code files in the `/.snippets/code/` directory, mirroring this file's path.
2. Reference snippets in your documentation using:

```
--8<-- "filename.ext"
```

For more features like only selecting a few lines from the file, check the docs:
https://facelessuser.github.io/pymdown-extensions/extensions/snippets/